### PR TITLE
fix: distinguish docker build errors from streaming errors

### PR DESCRIPTION
**Pull request by briandealwis**
In trying to reproduce #6126, I noticed that docker build errors are oddly prefixed with "`unable to stream build output:`".
```
Canceled build for leeroy-app
unable to stream build output: The command '/bin/sh -c exit 1' returned a non-zero code: 1. Please fix the Dockerfile and try again..
```

It turns out that the Docker code for parsing and interpreting the JSON message stream from the Docker daemon parses build failures as a `JSONError` object, which behaves like an `error`.  This PR changes our error-handling code to return such errors directly.

### CODE REVIEW

**Comment by briandealwis** [comment]
What's odd here is that this error can appear long after the output started.  Maybe `docker reports:` is a better prefix?

File: pkg/skaffold/docker/image.go [hunk 1]
```diff
@@ -364,6 +365,10 @@ func (l *localDaemon) Build(ctx context.Context, out io.Writer, workspace string
 	}
 
 	if err := streamDockerMessages(out, resp.Body, auxCallback); err != nil {
+		var jm *jsonmessage.JSONError
+		if errors.As(err, &jm) {
+			return "", err
```
Comment (briandealwis) [inline pkg/skaffold/docker/image.go:370]: Maybe this should be 
```suggestion
			return "", fmt.Errorf("docker build failure: %w", err)
```

Comment (tejal29) [reply pkg/skaffold/docker/image.go]: I like it better too.

**Review by tejal29 — APPROVED** [review]

**Comment by tejal29** [comment]
Not sure, why `blocked` label was added. Feel free to remove it!

**Comment by briandealwis** [comment]
I had it blocked pending the additional message.

**Pull request merged**